packages/p2p-v2/src/providers/AdvertiserInfoStateProvider/AdvertiserInfoStateProvider.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, PropsWithChildren, useContext } from 'react';
+import { TSocketError } from '@deriv/api-v2/types';
+
+type TContextValue = {
+    error: TSocketError<'p2p_advertiser_info'> | undefined;
+    isIdle: boolean;
+    isLoading: boolean;
+    isSubscribed: boolean;
+    setHasCreatedAdvertiser: (hasCreatedAdvertiser: boolean) => void;
+};
+
+const AdvertiserInfoStateContext = createContext<TContextValue>({} as TContextValue);
+
+type TAdvertiserInfoStateProviderProps = {
+    value: TContextValue;
+};
+
+/**
+ * A provider that contains the state of the current logged in user's advertiser info.
+ */
+export const AdvertiserInfoStateProvider = ({
+    children,
+    value,
+}: PropsWithChildren<TAdvertiserInfoStateProviderProps>) => (
+    <AdvertiserInfoStateContext.Provider value={value}>{children}</AdvertiserInfoStateContext.Provider>
+);
+
+/**
+ * Custom hook to access the current logged in user's advertiser info state.
+ * @returns {TContextValue} The current logged in user's advertiser info state.
+ */
+export const useAdvertiserInfoState = (): TContextValue => useContext(AdvertiserInfoStateContext);
```

Comment:
Because the state isn't cached or stored locally to the `useGetInfo` hook, the state will always be returned as the default values, which will affect how we try to get these states when trying to access it in different components when calling the hook. 

Because we need to know the state of the current logged in user, in a lot of different components, especially to determine if the user is an advertiser or not by checking for `AdvertiserNotFound` error, it is necessary to have this provider.